### PR TITLE
add UserResetPasswordCommand as PUT /api/rest/users/:id/reset

### PR DESCRIPTION
**Pull request by tgulacsi**
I need a global password reset ability.

### CODE REVIEW

File: core/commands/UserResetPasswordCommand.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+require_api( 'authentication_api.php' );
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'user_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that deletes a user account.
```
Comment (vboctor) [inline core/commands/UserResetPasswordCommand.php]: Update command description.  copy and paste error.

File: core/commands/UserResetPasswordCommand.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		if( $this->user_id_reset <= 0 ) {
+			throw new ClientException( 'Invalid user id', ERROR_INVALID_FIELD_VALUE, array( 'id' ) );
+		}
+
+		# Ensure user has access level to delete users
+		if( !access_has_global_level( config_get_global( 'manage_user_threshold' ) ) ) {
+			throw new ClientException( 'Access denied to reset user password', ERROR_ACCESS_DENIED );
+		}
+
+		$t_user = user_get_row( $this->user_id_reset );
```
Comment (vboctor) [inline core/commands/UserResetPasswordCommand.php]: I assume you tested resetting users that don't exist.  I can see the user_get_row() throwing so it should work, but test to make sure all is working as expected.

File: core/commands/UserResetPasswordCommand.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		if( !access_has_global_level( $t_user['access_level'] ) ) {
+			throw new ClientException( 'Access denied to reset user password with higher access level', ERROR_ACCESS_DENIED );
+		}
+
+		# Check that we are not reseting the last administrator account
+		$t_admin_threshold = config_get_global( 'admin_site_threshold' );
+		if( user_is_administrator( $this->user_id_reset ) &&
+			user_count_level( $t_admin_threshold, /* enabled */ true ) <= 1 ) {
+			throw new ClientException(
+				'Reseting last administrator not allowed',
```
Comment (vboctor) [inline core/commands/UserResetPasswordCommand.php]: Typo: `Resetting`

**Comment by vboctor** [comment]
Please also create a feature request (under `api rest` category) in the bug tracker and reference it from the change commit(s).  For example:

```
Support user password reset via REST API

Fixes #123456
```

**Comment by dregad** [comment]
@tgulacsi are you planning to address the review comments ?

**Comment by tgulacsi** [comment]
Yes, sorry.

**Comment by tgulacsi** [comment]
It's been reported as  https://mantisbt.org/bugs/view.php?id=26632 .

**Comment by dregad** [comment]
Thanks @tgulacsi. 
@vboctor let me know if you're happy with the fixes before merge.

File: manage_user_reset.php [hunk 4]
```diff
@@ -56,20 +56,12 @@
 
 user_ensure_exists( $f_user_id );
 
-$t_user = user_get_row( $f_user_id );
+$t_data = array(
```
Comment (vboctor) [inline manage_user_reset.php]: Remove the access check for manager user threshold and user ensure exist checks above.  Remove require statements that are not longer necessary.

**Comment by dregad** [comment]
@tgulacsi your PR introduced a regression, it is no longer possible to reset a user's password from the GUI.

Tracking: [#26880](https://mantisbt.org/bugs/view.php?id=26880)

**Pull request merged**